src/main/java/com/google/sps/data/Room.java

```diff
@@ -139,6 +127,18 @@ private Room(List<Member> members, Queue<Video> videos, Queue<Message> messages,
         this.roomId = id;
     }
 
+    /**
+      * Room constructor
+      * @param members a List of Member objects
+      * @return a new Room object
+      */
+    public Room(List<Member> members) {
+        this.members = members;
+        this.videos = new LinkedList<Video>();
+        this.messages = new LinkedList<Message>();
+        this.roomId = null;
```

Comment:
What's the use case of this function? Why roomId can be optional in a constructor?

Reply:
Because on Room creation, the room Id is not necessary because it has not been stored in datastore before.

Reply:
This might be a design question. Room id is supposed to be a id set during room creation time no matter as a required field.

Reply:
Datastore does this for us, which is why it is set as null during room creation, but afterwards when it is retrieved it is a value